source/Server.Extensibility/TinyTypes/TinyType.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+
+namespace Octopus.Server.Extensibility.TinyTypes
+{
+    /// <summary>
+    ///     Provides a strongly-typed wrapper around a (ideally value) type.
+    /// </summary>
+    /// <remarks>
+    ///     NOTE: We deliberately do not provide implicit or explicit cast operators. Implicit cases would defeat the purpose
+    ///     of providing compile-time errors when disparate types are compared or passed as parameters. Explicit cast operators
+    ///     would not cause this problem... but they would be *so very tempting* that it's inevitable that they'd end up being
+    ///     used to coerce types between values that really shouldn't be coerced. The design decision at present is that tiny
+    ///     types must be explicitly unwrapped and re-wrapped in order to minimise the likelihood of bugs' being created.
+    /// </remarks>
+    public abstract class TinyType<T>
```

Comment:
'ang on, isn't TinyTypes a library? Why is it inlined here?

Reply:
Oops - I should have answered this question first. Copy/paste from my comment above:

I'd expect that we'd pull these out into the TinyTypes library soon-ish but there are some design differences (mainly around implicit cast operators) that I'd like to discuss first. I didn't want to introduce yet another dependency in the chain that would be so far removed from where we're actually trying to use it, which is already one consumer upstream from here.